native/src/seal/polyarray.cpp

```diff
@@ -52,6 +52,29 @@ namespace seal
         }
     }
 
+    PolynomialArray &PolynomialArray::operator=(const PolynomialArray &assign)
+    {
+        // Check for self-assignment
+        if (this == &assign)
+        {
+            return *this;
```

Comment:
If a type is mutable, this is very dangerous.

Reply:
This is what the other classes do.